src/snowflake/cli/plugins/sql/manager.py

```diff
@@ -35,9 +35,30 @@ def execute(
 
         if std_in:
             query = sys.stdin.read()
-        elif file:
-            query = SecurePath(file).read_text(file_size_limit_mb=UNLIMITED)
+        elif files and len(files) == 1:
+            # If there's only one file we need to handle case of single query
+            query = SecurePath(files[0]).read_text(file_size_limit_mb=UNLIMITED)
```

Comment:
Non-blocking: I'd suggest removing the special case for a single file and instead tally all queries you ever found to figure out if there is a "single statement" or not, but maybe you are working around some property of `split_statements` (does it erroneously return non-statements if a file is empty of non-comment SQL?).

Reply:
Good point, changed